analyses/chromosomal-instability/01-localization-of-breakpoints.Rmd

```diff
@@ -0,0 +1,279 @@
+      start_col = "coord",
+      end_col = "coord",
+      window_size = bin_size, # Bin size to calculate breaks density
+      chr_sizes_vector = chr_sizes_vector, # This is the sizes of chromosomes used for binning
+      unsurveyed_bed = uncallable_bed, # This is the BED file that notes what regions are uncallable
+      perc_cutoff = .75, # What percentage of each bin needs to be callable for it not to be NA
+      return_vector = return_vector # If we don't need the full GenomicRanges object
+    )
+}
+```
+
+### Read in data 
+
+Import data that was set up in breakpoint format in `00-setup-breakpoint-data.R`.
+
+```{r}
+breaks_list <- readr::read_rds(file.path(output_dir, "breaks_lists.RDS"))
+```
+
+Get the list of biospecimen IDs that are in this set.
+
+```{r}
+common_samples <- unique(breaks_list$intersection_of_breaks$samples)
+```
+
+Set up metadata
+
+```{r}
+# Read in the metadata
+metadata <- readr::read_tsv(file.path(data_dir, "pbta-histologies.tsv")) %>% 
+  # Only keep metadata for the samples we are working with here
+  dplyr::filter(Kids_First_Biospecimen_ID %in% common_samples)
+```
+
+Set up chromosome size data that will be used for creating genome bins.
+It just so happens that this BED file: `WGS.hg38.strelka2.unpadded.bed` is actually 
+just a list of the chromosome sizes so we are using that for now. 
+
+```{r}
+# Set up Chr sizes
+chr_sizes_bed <- readr::read_tsv(file.path(data_dir, "WGS.hg38.strelka2.unpadded.bed"),
+  col_names = c("chrom", "start", "end")
+) %>%
+  # Reformat the chromosome variable to drop the "chr"
+  dplyr::mutate(chrom = factor(gsub("chr", "", chrom),
+    levels = c(1:22, "X", "Y", "M")
+  )) %>%
+  # Remove sex chromosomes
+  dplyr::filter(!(chrom %in% c("X", "Y", "M")))
+
+# Make chromosome size named vector for Heatmap annotation
+chr_sizes_vector <- chr_sizes_bed$end
+names(chr_sizes_vector) <- chr_sizes_bed$chrom
+```
+
+Set up uncallable regions BED. We will use this to declare NAs in regions with
+too high of a percentage of uncallable regions. 
+
+```{r}
+uncallable_bed <- readr::read_tsv(
+  file.path(
+    "..",
+    "copy_number_consensus_call",
+    "ref",
+    "cnv_excluded_regions.bed"
+  ),
+  col_names = c("chrom", "start", "end")
+) %>%
+  # Reformat the chromosome variable to drop the "chr"
+  dplyr::mutate(chrom = factor(gsub("chr", "", chrom),
+    levels = c(1:22, "X", "Y")
+  )) %>%
+  dplyr::filter(
+    !is.na(chrom),
+    # Drop sex chromosomes
+    !(chrom %in% c("X", "Y", "M"))
+  )
+```
+
+### Mapping of breakpoint densities for individual samples 
+
+Let's reformat that the intersection_or_breaks data.frame. 
+It contains both SV data and CNV data for its intersections. 
+We'll go by SV data here. 
+
+```{r}
+# For simplicity's sake, change the name of the intersection column we want to use
+breaks_list$intersection_of_breaks <- breaks_list$intersection_of_breaks %>%
+  dplyr::rename(
+    chrom = sv_ranges.seqnames,
+    coord = sv_ranges.start
+  )
```

Comment:
Yes. But also I'm not happy with my "solution" so if you have a better one, I'd like to use it. When I implemented the gap option, this means the coordinates that are intersecting here are not *exactly* identical between CNV and SV data. Which means we have two sets of coordinates and here I've arbitarily chose to use the SV coordinates to represent the intersection data. But I could have just as easily use the CNV coordinates. But I don't want to get rid of these column names when I originally make the data because I don't want this information to get lost.

Reply:
That makes sense. I guess if there were a reason that the positions were not the same, you could calculate the average of the start and end positions, but that seems like it is probably overkill.